phoenix-core/src/it/java/org/apache/phoenix/end2end/InListIT.java

```diff
@@ -43,27 +46,39 @@
 import org.apache.phoenix.schema.types.PInteger;
 import org.apache.phoenix.util.PhoenixRuntime;
 import org.apache.phoenix.util.PropertiesUtil;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 
-import com.google.common.base.Function;
-import com.google.common.base.Joiner;
-import com.google.common.collect.Lists;
-
 
 public class InListIT extends ParallelStatsDisabledIT {
     private static final String TENANT_SPECIFIC_URL1 = getUrl() + ';' + TENANT_ID_ATTRIB + "=tenant1";
-    String tableName;
-    String descViewName;
-    String ascViewName;
+    private static boolean isInitialized = false;
+    private static String tableName = generateUniqueName();
+    private static String tableName2 = generateUniqueName();
+    private static String descViewName = generateUniqueName();
+    private static String ascViewName = generateUniqueName();
+    private static String viewName1 = generateUniqueName();
+    private static String viewName2 = generateUniqueName();
+    private static String prefix = generateUniqueName();
 
     @Before
     public void setup() throws Exception {
-        tableName = generateUniqueName();
```

Comment:
Instead of the initialized variable, could we change this to the @BeforeClass annotation instead?
In fact would the whole test still work if we only change @Before to @BeforeClass?

Reply:
And sorry I missed this for so long. I was off end of July and then missed it.

Reply:
> would the whole test still work if we only change @before to @BeforeClass?
Yeah , actually @BeforeClass requires a method to be static so all the data structures need to be updated as static. I didn't spend time doing that probably because of being lazy though could have used @BeforeAll and make the Junit lifecycle to be @TestInstance(Lifecycle.PER_CLASS) to avoid an initialized variable.

Reply:
I looked up @TestInstance because it sounds interesting, but it's a Junit5 feature, and we're still on Junit 4. AFAICT The only solutions are your current one, or using @BeforeClass as you said.